podman-compose: add page (#8726)
tldr-pages/tldr: pages/common/podman-compose.md

# podman-compose

> > Run and manage compose spec container definition.

Reviewer: Base on the official webpage we can use the full name
```suggestion
> Run and manage Compose Specification container definition.
```

> > More information: <https://podman.readthedocs.io/>.

Reviewer 2: ```suggestion
> More information: <https://github.com/containers/podman-compose>.
```
I couldn't find documentation in the mentioned link. Seems like it can only be accessed using the `--help` command. So I think it is better to update to their GitHub repository.

Adding the output here for future reference

```bash
usage: podman-compose [-h] [-v] [-f file] [-p PROJECT_NAME]
                      [--podman-path PODMAN_PATH] [--podman-args args]
                      [--podman-pull-args args] [--podman-push-args args]
                      [--podman-build-args args] [--podman-inspect-args args]
                      [--podman-run-args args] [--podman-start-args args]
                      [--podman-stop-args args] [--podman-rm-args args]
                      [--podman-volume-args args] [--no-ansi] [--no-cleanup]
                      [--dry-run]
                      {help,version,pull,push,build,up,down,ps,run,exec,start,stop,restart,logs}
                      ...

options:
  -h, --help            show this help message and exit
  -v, --version         show version
  -f file, --file file  Specify an alternate compose file (default: docker-compose.yml)
  -p PROJECT_NAME, --project-name PROJECT_NAME
                        Specify an alternate project name (default: directory name)
  --podman-path PODMAN_PATH
                        Specify an alternate path to podman (default: use location in $PATH variable)
  --podman-args args    custom global arguments to be passed to `podman`
  --podman-pull-args args
                        custom arguments to be passed to `podman pull`
  --podman-push-args args
                        custom arguments to be passed to `podman push`
  --podman-build-args args
                        custom arguments to be passed to `podman build`
  --podman-inspect-args args
                        custom arguments to be passed to `podman inspect`
  --podman-run-args args
                        custom arguments to be passed to `podman run`
  --podman-start-args args
                        custom arguments to be passed to `podman start`
  --podman-stop-args args
                        custom arguments to be passed to `podman stop`
  --podman-rm-args args
                        custom arguments to be passed to `podman rm`
  --podman-volume-args args
                        custom arguments to be passed to `podman volume`
  --no-ansi             Do not print ANSI control characters
  --no-cleanup          Do not stop and remove existing pod & containers
  --dry-run             No action; perform a simulation of commands

command:
  {help,version,pull,push,build,up,down,ps,run,exec,start,stop,restart,logs}
    help                show help
    version             show version
    pull                pull stack images
    push                push stack images
    build               build stack images
    up                  Create and start the entire stack or some of its services
    down                tear down entire stack
    ps                  show status of containers
    run                 create a container similar to a service to run a one-off command
    exec                execute a command in a running container
    start               start specific services
    stop                stop specific services
    restart             restart specific services
    logs                show logs from services
```

Author: Done :)

Reviewer 2: > Done :)

Thanks, Don't resolve this conversation. This output can be referred to by other reviewers.

> I will review this tomorrow.

- List all running containers:

`podman-compose ps`

- Create and start all containers in the background using a local `docker-compose.yml`:

`podman-compose up -d`

- Start all containers, building if needed:

`podman-compose up --build`

- Start all containers using an alternate compose file:

`podman-compose {{path/to/file}} up`

- Stop all running containers:

`podman-compose stop`

- Remove all containers, networks, and volumes:

`podman-compose down --volumes`

> - Follow logs for a containers (omit container name for all):

Reviewer 2: ```suggestion
- Follow logs for containers (omit all container names):
```

> - Follow logs for a containers (omit container name for all):

Reviewer 2: ```suggestion
- Follow logs for a container (omit all container names):
```
I think this would be better.

`podman-compose logs --follow {{container_name}}`

> - Run a one-time command in a service (no ports are mapped):

Reviewer 2: ```suggestion
- Run a one-time command in a service with no ports mapped:
```
Would this be better?
